packages/notifi-react-card/lib/components/subscription/NotifiSubscriptionCard.tsx

```diff
@@ -61,61 +59,12 @@ export type NotifiSubscriptionCardProps = Readonly<{
 
 export const NotifiSubscriptionCard: React.FC<
   React.PropsWithChildren<NotifiSubscriptionCardProps>
-> = ({
-  classNames,
-  cardId,
-  darkMode,
-  inputLabels,
-  inputs = {},
-  inputSeparators,
-  children,
-}: React.PropsWithChildren<NotifiSubscriptionCardProps>) => {
-  const { isInitialized } = useNotifiSubscribe();
-  const { loading } = useNotifiSubscriptionContext();
-  const inputDisabled = loading || !isInitialized;
-
-  const card = useSubscriptionCard(cardId);
-  let contents: React.ReactNode = null;
-
-  switch (card.state) {
-    case 'loading':
-      contents = (
-        <LoadingStateCard
-          classNames={classNames?.LoadingStateCard}
-          card={card}
-        />
-      );
-      break;
-    case 'error':
-      contents = (
-        <ErrorStateCard classNames={classNames?.ErrorStateCard} card={card} />
-      );
-      break;
-    case 'fetched':
-      contents = (
-        <FetchedStateCard
-          classNames={classNames?.FetchedStateCard}
-          card={card}
-          inputs={inputs}
-          inputDisabled={inputDisabled}
-          inputLabels={inputLabels}
-          inputSeparators={inputSeparators}
-        />
-      );
-      break;
-  }
+> = (props: React.PropsWithChildren<NotifiSubscriptionCardProps>) => {
+  const { params } = useNotifiClientContext();
 
   return (
-    <div
-      className={clsx(
-        darkMode ? 'notifi__dark' : 'notifi__light',
-        'NotifiSubscriptionCard__container',
-        classNames?.container,
-      )}
-    >
-      {children}
-      {contents}
-      <NotifiFooter classNames={classNames?.NotifiFooter} />
-    </div>
+    <NotifiSubscriptionContextProvider {...params}>
+      <NotifiSubscriptionCardContainer {...props} />
```

Comment:
I'm not very good at naming, let me know if you could figure out a better name than NotifiSubscriptionCardContainer 😂.